packages/execution-environments/src/common/keyring.ts

```diff
@@ -0,0 +1,24 @@
+import { SnapKeyring } from '@metamask/snap-types';
+
+/**
+ * Wraps a SnapKeyring class, returning a handler that can route requests to the exposed functions by the class.
+ *
+ * @param keyring - The SnapKeyring instance.
+ * @returns A handler function.
+ */
+export function wrapKeyring(keyring?: SnapKeyring) {
+  if (!keyring) {
+    throw new Error('Keyring not exported');
+  }
+
+  const keyringHandler = ({ request }: any) => {
```

Comment:
Can we use a proper type for this?

Reply:
Yeah, we should. I have yet to clean up the types in general 😅